homeassistant/components/numato/binary_sensor.py

```diff
@@ -0,0 +1,98 @@
+"""
+Binary sensor platform integration for Numato USB GPIO expanders.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/integrations/numato#binary-sensor
+"""
+import logging
+
+from numato_gpio import NumatoGpioError
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import DEVICE_DEFAULT_NAME
+
+from . import CONF_BINARY_SENSORS, CONF_ID, DOMAIN, edge_detect, read_input, setup_input
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_INVERT_LOGIC = "invert_logic"
+CONF_PORTS = "ports"
+CONF_DEVICES = "devices"
+CONF_DEVICE_ID = "id"
+DEFAULT_INVERT_LOGIC = False
+
+DEPENDENCIES = ["numato"]
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the configured Numato USB GPIO binary sensor ports."""
+
```

Comment:
Add a guard clause here that checks if discovery_info is None and return if so.

Reply:
discovery_info is usually None when setup_platform is called. Probably because I'm not using Home-Assistant's discovery functionality/API at the moment. I probably should dive into this, too, should I?

Reply:
Pass at least an empty dict as fourth argument in `load_platform`. That will be the discovery_info.

Reply:
Done.